packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -1336,6 +1336,67 @@ export interface IntelligentTieringConfiguration {
   readonly deepArchiveAccessTierTime?: Duration;
 }
 
+/**
+ * The date source for the partitioned prefix.
+ */
+export enum PartitionDateSource {
+  /**
+   * The year, month, and day will be based on the timestamp of the S3 event in the file that's been delivered.
+   */
+  EVENT_TIME = 'EventTime',
+
+  /**
+   * The year, month, and day will be based on the time when the log file was delivered to S3.
+   */
+  DELIVERY_TIME = 'DeliveryTime',
+}
+
+/**
+ * The key format for the log object.
+ */
+export abstract class TargetObjectKeyFormat {
+  /**
+   * Use partitioned prefix for log objects.
+   *
+   * The partitioned prefix format as follow:
+   * [DestinationPrefix][SourceAccountId]/​[SourceRegion]/​[SourceBucket]/​[YYYY]/​[MM]/​[DD]/​[YYYY]-[MM]-[DD]-[hh]-[mm]-[ss]-[UniqueString]
+   */
+  public static partitionedPrefix(dateSource?: PartitionDateSource): TargetObjectKeyFormat {
```

Comment:
is the `dataSource` parameter optional here? whats the behaviour if they do 
```
targetObjectKeyFormat: s3.TargetObjectKeyFormat.partitionedPrefix()
```

Reply:
@aaythapa 
Thanks for clarifying.
If the `dataSource` argument is not specified, `EventTime` is used as a default date source.

> PartitionedPrefix defaults to EventTime delivery when server access logs are delivered.
> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-partitionedprefix.html

Is it better to explicitly set `EventTime` for `CfnBucket` if no argument is specified?

Reply:
@sakurai-ryo I think it'll depend. If the user doesn't set `dataSource` then the CFN template will look like the following (let me know if I'm wrong)
```
TargetObjectKeyFormat:
    PartitionedPrefix:
       PartitionDateSource: undefined
```
Will this still deploy and set the delivery to `EventTime`? Or is there a deployment error? If there is a deployment error then I think we should explicitly set it, if not I think it's fine to leave it as is.

Either way we should state somewhere (whether in the doc-string or the readme) that `PartitionedPrefix` defaults to `EventTime` delivery

Reply:
@aaythapa 
Thanks for digging into this and sorry for my lack of explanation.
In this case, the template would look like this.
I think this is because the undefined property disappears when stringifying from JS to JSON.
So the deployment will succeed.
I added integ-test to verify this.

```json
  "MyBucket43E0A113B": {
   "Type": "AWS::S3::Bucket",
   "Properties": {
    "LoggingConfiguration": {
     "DestinationBucketName": {
      "Ref": "MyAccessLogsBucketF7FE6635"
     },
     "LogFilePrefix": "example4",
     "TargetObjectKeyFormat": {
      "PartitionedPrefix": {}
     }
    }
   },
   "UpdateReplacePolicy": "Delete",
   "DeletionPolicy": "Delete"
  }
```

> Either way we should state somewhere (whether in the doc-string or the readme) that PartitionedPrefix defaults to EventTime delivery

Absolutely right.
I added the comment in doc-string.

Reply:
Awesome, thank you